src/minimega/container.go

```diff
@@ -1002,10 +1002,12 @@ func (vm *ContainerVM) launch() error {
 
 			// containers cannot return unless thawed, so thaw the
 			// process if necessary
+			vm.lock.Lock()
 			if err = vm.thaw(); err != nil {
 				log.Errorln(err)
 				vm.setError(err)
 			}
+			vm.lock.Unlock()
```

Comment:
Hm, I think we want to defer the unlock immediately after locking, right? There are calls to `vm.setError` below this that assume that the caller has acquired the VM's lock. I don't see anything in the remainder of the goroutine that tries to reacquire the lock.